src/audio/module_adapter/module_adapter.c

```diff
@@ -1275,12 +1278,12 @@ void module_adapter_free(struct comp_dev *dev)
 	mod_free_all(mod);
 
 #if CONFIG_IPC_MAJOR_4
-	rfree(mod->priv.cfg.input_pins);
+	module_driver_heap_free(drv->user_heap, mod->priv.cfg.input_pins);
 #endif
 
-	rfree(mod->stream_params);
-	rfree(mod);
-	rfree(dev);
+	module_driver_heap_free(drv->user_heap, mod->stream_params);
+	module_driver_heap_free(drv->user_heap, mod);
+	module_driver_heap_free(drv->user_heap, dev);
```

Comment:
Fwiw, I have a change coming here that will mean some abstraction. Yes we will still be able to assign the pages to the user driver domain, but the method to allocate/free will be different as pipeline/DP modules will have virtual continuous pages preallocated for them (and the driver user heap would reside in these pages).